configuration/config2/SchemaType.hpp

```diff
@@ -37,9 +37,13 @@ class DECL_EXPORT SchemaType
 
         std::shared_ptr<SchemaTypeLimits> &getLimits() { return m_pLimits; }
         void setLimits(const std::shared_ptr<SchemaTypeLimits> &pLimits) { m_pLimits = pLimits; }
-        bool isValid() const { return m_pLimits!=nullptr; }
+        bool isValid() const { return m_pLimits!=nullptr || m_baseType.empty(); }
```

Comment:
this might be written as intended, but at a quick glance the logic seems a bit odd, just make sure logic is  correct.
Update: I see a isValueValid method below, consider making this method explicitly self documenting as well...